.github/workflows/main_debug.yml

```diff
@@ -1,27 +1,30 @@
 name: CI-Debug
 
-on:
+on: 
+# Runs on every push on master branch. If a push contains multiple commits, it will be ran on the latest one.
   push:
     # Runs on every push on master branch
     branches:
       - master
   pull_request:
     paths-ignore:
       - 'doc/**'
-
+    
 # Cancels running instances when a pull request is updated by a commit
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref || github.run_id }}
   cancel-in-progress: true
 
+
 env:
-  COMPILE_JOBS: 2
-  MULTI_CORE_TESTS_REGEX: "mpirun=2"
+  COMPILE_JOBS: 18
+  TEST_JOBS: 12
```

Comment:
Why 12?

Reply:
Threading seems to make the test run slower on Maxwell. Threading seems good for compiling but not for testing.

Reply:
Sacré Maxwell